backend/runner/taskrun/data_update_executor.go

```diff
@@ -254,5 +273,9 @@ func (exec *DataUpdateExecutor) backupData(
 			)
 		}
 	}
-	return nil
+
+	if len(priorBackupDetail.Items) == 0 {
```

Comment:
Any reason not returning priorBackupDetail?

Reply:
We can remove "&& priorBackupDetail != nil" on line 109 as well.

Reply:
> Any reason not returning priorBackupDetail?

Nothing is backed up, so it's better to pass a `nil`.

Reply:
There isn't much difference between &{} and nil (nil or []string{}) from go proto perspective. It's totally fine to return &{} here.

Or we can return nil, nil early on line 109 if len(statement) == 0.

Reply:
Done